lib/player.js

```diff
@@ -2247,7 +2249,14 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     }
 
     // Re-filter the manifest after streams have been chosen.
-    this.filterManifestByCurrentVariant_();
+    // Note: Skip if an audio codec switching strategy is being enforced.
+    if (this.config_.streaming.codecSwitchingStrategy ===
+      shaka.config.CodecSwitchingStrategy.DISABLED) {
+      this.filterManifestByCurrentVariant_();
+    } else {
+      shaka.log.debug('Skipping re-filter. Not necessary' +
```

Comment:
Here I think it is necessary to do a more complex filtering...

Imagine the next stream:

Video:
360p - h264 - 500kbps
720p - h264 - 1000kbps
720p - h265 - 800kbps
1080p - h265 - 1500kbps

Audio:
en - aac - 2 channels - 128kbps
es - ac3 - 2 channels - 128kbps
es - ac3 - 6 channels - 320kbps
es - aac - 6 channels - 256kbps

What would be expected to return (I would like @joeyparrish to confirm)

Video:
360p - h264 - 500kbps
720p - h265 - 800kbps
1080p - h265 - 1500kbps

Audio:
en - aac - 2 channels - 128kbps
es - ac3 - 2 channels - 128kbps
es - aac - 6 channels - 256kbps

Reply:
Going to discuss this with @sky-hugolima in the morning. Looking to close out remaining change requests; before re-addressing the filtration logic.

Reply:
Hey @avelad, sorry for the delay replying on this comment, we had to spend some time refactoring the actual switching logic to make it work correctly for the reload strategy.

Looking at your comment above, it makes sense, and while we actually don't experience those scenarios on our particular Peacock streams, we do understand that's a a valid concern.

I think where these changes affect the intended behaviour tho, is not specifically on not calling `filterManifestByCurrentVariant_` because that method's single purpose is to filter-out variants that have codecs incompatible with the currently selected variant, and that logic doesn't apply if we can switch codecs, but instead the place you mention bellow `chooseCodecsAndFilterManifest` is where I think we need to make some other changes.

It's going to be a somewhat more complex refactoring to the `chooseCodecsAndFilterManifest` logic to output the intended result, and filter-out the variants that conflict at the same audio/video specs, but not variants where the non optimal codecs are the only ones available for that particular audio/video spec.

Please advise if the above assumption is incorrect.

Reply:
That makes sense to me.  I haven't finished reviewing the entire PR, but filtering by current variant should be skipped AFAICT.  Removal of redundant streams is a separate issue, I think.